README.md
# Aplikasi-Mapserver-TB-GIS

Hello guysss ...

Ini adalah aplikasi mapserver sederhana untuk menampilkan peta lokasi dari kelompok TB GIS Mapserver yaitu:

1. Rosnani Nasrul                 (1134015)

2. Nur Mulyana Mustari            (1134047)

3. Nur Fatimah Hasan              (1134016)

4. Hariyati Suarni Nurindah       (1134021)

Kami dari jurusan D4 Teknik Informatika Kelas 3B Angkatan 2013

Semoga aplikasi ini dapat bermanfaat bagi pembaca :) 

Terima Kasih Kunjungannya ^_^


ms4w/HISTORY.txt
MS4W Release History
====================

2012-05-26 Gateway Geomatics
  * ms4w version 3.0.6 created. Notable changes are:
         - rebuilt cairo, hdf5, libxml2 due to reliance on old
           zlib version (ticket 2394)

2012-05-25 Gateway Geomatics
  * ms4w version 3.0.5 created. Notable changes are:
         - upgraded MapServer and MapScript to 6.0.3 (ticket 2393)  
         - upgraded GDAL to 1.9.1 (ticket 2387)  
         - upgraded PHP to 5.4.3 (ticket 2368)
         - security: - moved php.exe from the cgi-bin to the /tools 
                       directory
                     - moved \Apache\htdocs\phpinfo.php to \tmp
                     - thank you to Mike Arnold
         - upgraded Apache to 2.2.22, with OpenSSL 0.9.8t (ticket 2392)  
         - upgraded mod_fcgid to 2.3.7
         - upgraded the CA bundle file in \Apache\conf\ca-bundle\
         - upgraded to jpeg 8d support for GDAL and MapServer 
         - upgraded to zlib 1.2.7 support for GDAL and MapServer  
         - upgraded to libPNG 1.5.10 support for GDAL and 
           MapServer 
         - upgraded to libTIFF 4.0.1 support for GDAL
         - upgraded to GeoTIFF 1.4.0 support for GDAL (ticket 2391)
         - upgraded to expat 2.1.0 support for GDAL
         - upgraded to MrSID SDK 8.5 support for GDAL
         - upgraded to PostgreSQL 9.1.3 support for GDAL
         - upgraded PROJ.4 to 4.8.0 (ticket 2366)
         - upgraded to Oracle SDK 11.2.0.3.0 for GDAL and MapServer 
         - upgraded to ESRI File Geodatabase 1.2 API support for GDAL
         - upgraded to OpenSSL 1.0.1c support for GDAL
         - upgraded to CURL 7.25.0 support for GDAL and MapServer  
         - upgraded to GEOS 3.3.3 support for GDAL and 
           MapServer (ticket 2377) 
         - upgraded to libKML svn-05242012 support for GDAL
         - upgraded to Freetype 2.4.9 support for MapServer
         - replaced GNU RegEx library with PCRE library (ticket 2365)
         - upgraded to libxml2 2.7.8 support for MapServer         

2012-02-13 Gateway Geomatics
  * ms4w version 3.0.4 created. Notable changes are:
         - upgraded PHP to 5.3.10 (ticket 2362)
         - upgraded MapServer and MapScript to 6.0.2 release (ticket 2364)

2012-01-22 Gateway Geomatics
  * ms4w version 3.0.4-beta1 created. Notable changes are:
         - added MapCache module to Apache (ticket 2306)
         - upgraded GDAL to 1.9.0 (ticket 2356)   
         - upgraded Apache to 2.2.21, with OpenSSL 0.9.8r (ticket 2349)
         - upgraded PHP to 5.3.9 (ticket 2343)
         - rebuilt PHP with LDAP support (ticket 2353)         
         - added Apache and PHP licenses to MS4W Component 
           Licensing (ticket 2310) 
         - added ESRI File Geodatabase support for GDAL (ticket 2338)
         - upgraded to GEOS 3.3.2 support for GDAL and 
           MapServer (ticket 2357)
         - upgraded to libPNG 1.5.7 support for GDAL and 
           MapServer (ticket 2287)
         - upgraded to libTIFF 4.0.0 support for GDAL (ticket 2358)
         - upgraded to geoTIFF 1.3.1-dev support for GDAL
         - upgraded to PostgreSQL 9.1.2 support for GDAL
         - upgraded to Freetype 2.4.8 support for MapServer
         - upgraded to Oracle SDK 11.2.0.2.0 for GDAL and MapServer
         - upgraded to Poppler 0.20.0 support for GDAL         

2011-07-12 Gateway Geomatics
  * ms4w version 3.0.3 created. Notable changes are:
         - upgraded MapServer and MapScript to 6.0.1 security release 
           (ticket 2336)  
         - upgraded GDAL to 1.8.1RC2 (ticket 2337)    
         - upgraded Apache to 2.2.19, with OpenSSL 0.9.8r (ticket 2325)
         - added libpoppler library to GDAL build to read geospatial 
           PDF documents (ticket 2308)
         - added Cairo library for PDF and SVG output in MapServer
         - added giflib library to MapServer build
         - added missing Python scripts into /python/gdal/ (ticket 2317)
         - removed extra SetEnv PATH from Apache's httpd.conf (ticket 2312)
         - added VERSION.txt document to MS4W root (ticket 2309)

2011-04-08 Gateway Geomatics
  * ms4w version 3.0.2 created. Notable changes are:
         - upgraded PHP to 5.3.6 (ticket 2299)
         - added missing CURL_CA_BUNDLE to Apache (ticket 2305)
         - added libKML support for GDAL (ticket 2294)

2011-01-27 Gateway Geomatics
  * ms4w version 3.0.1 created. Notable changes are:
         - upgraded MapServer and MapScript to 5.6.6 release 
           (ticket 2286)  
         - upgraded GDAL to 1.8.0 (ticket 2289)  
         - upgraded to MrSID SDK 8.0 support for GDAL (ticket 2291)
         
2011-01-11 Gateway Geomatics
  * ms4w version 3.0 created. Notable changes are:
         - upgraded MapServer and MapScript to 5.6.5 release 
           (ticket 2246)  
         - upgraded GDAL to 1.7.3 (ticket 2280)  
         - upgraded Apache to 2.2.17 (ticket 2279)
         - upgraded PHP to 5.3.5 (ticket 2278)
         - rebuilt PHP with exif support (ticket 2261)
         - upgraded mod_fcgid to 2.3.6 (ticket 2281)
         - fixed GIF symbol rendering problem based on size 
           (ticket 2094)
         - upgraded to PostgreSQL 9.0.2 support for GDAL         
         - upgraded to jpeg 8b support for GDAL and MapServer
         - upgraded to zlib 1.2.5 support for GDAL and MapServer
         - upgraded to geotiff 1.3.1-dev support for GDAL
         - upgraded to xerces 3.1.1 support for GDAL
         - upgraded to HDF5 5.1.8 support for GDAL
         - upgraded to OpenSSL 1.0.0c support for GDAL
         - upgraded to CURL 7.21.3 support for GDAL and MapServer
         - upgraded to Freetype 2.4.4 support for MapServer
         - upgraded to Iconv 1.11.1 support for MapServer
         - upgraded to libxml2 2.7.7 support for MapServer
         - added shptreetst utility (ticket 2253)

2010-04-20 Gateway Geomatics
  * ms4w version 3.0-beta11 created. Notable changes are:
        - upgraded Apache to 2.2.15 (ticket 2178)

webs/Login.php
<?
ini_set('display_errors',false);
if (isset($_POST['login']))
{
include "./include/conn.php";
$username=htmlentites((trim($_POST['username'])));
$password=htmlentites(md5($_POST['password']));

$login=mysql_db_query($db,"select * from daftar where user='$username' and pass='$password'",$koneksi);
while($row=mysql_fetch_array($login))
{
$user_id=$row[0];
$user_name=$row[2];
$status=$row['status'];
}

$cek_login=mysql_num_rows($login);
if (empty($cek_login))
{
?>
<script languange="javascript">alert("password atau username anda salah!!");</script>
<script> document.location.href='index.php?page=4'; </script>
<?
}
else
{
  //ubah status login
  $update_stat=mysql_db_query($db,"update daftar set status='1' where id='$user_id'",$koneksi);
  session_register('user_id');
  session_register('user_name');
  ?>
  <script languange="javascript">allert("anda berhasil login");</script>
  <script> document.location.href='index.php?page=4'; </script>
  <?
}

}
?>

<html>
  <script languange="javascript">
  function cek(){
    var user=document.getElementById('userid').value;
    var pass= document.getElementById('passwd').value;
	if(user.replace(/^\s+|\s+$/g, '')==''){
		alert('Maaf, Username Anda masih kosong !');
		return false;
	} 
	if(pass.replace(/^\s+|\s+$/g, '')==''){
		alert('Maaf, Password Anda masih kosong  !');
		return false;
	}
	return true;
}
</script>
  }
  </script>
<table width="30%" border="0" cellpadding="0" cellspacing="0" bordercolor="#99CC99">
<tr> 
	<td bgcolor="#FF0000" width="6%">&nbsp;</td>
	<td width="70%" bgcolor="#FF0000" ><div align="center"><strong><font face="times new roman" size="4" color="#FFFFFF">Login</font></strong></div></td>
	<td width="6%" bgcolor="#FF0000">&nbsp;</td>
</tr>
<tr>
	<td bgcolor="#FF0000">&nbsp;</td>
	<td>
	<table width="250" align="center">
		<tr><td width="157"><font face="times new roman" size="4">&nbsp;
		</font>
		    
		<form name="masuk" action="login.php" method="post">
		<table border="0" bordercolor="red" align="left">
	<tr>	
		<td><font face="verdana" size="2">Username</font></td>
		<td><input type="text" size="10" name="username" id="userid"></td>
	</tr>
	
	<tr>
		<td><font face="verdana" size="2">Password</font></td>
		<td><input type="password" size="10" name="password" id="passwd"></td>
	</tr>
	<tr>
		<td><input type="submit" value="Login" name="login" onClick=" return cek()"></td>
		<td>
		<input type=button value="SignUp" onClick="window.open('signup.php','scrollwindow','top=200,left=350,width=575,height=400');">
		</td>
	</tr>
</table>
		</form>
		
		</td></tr>
	</table>
	</td>
	<td bgcolor="#FF0000">&nbsp;</td>
	<td width="2%"></td>
</tr>
<tr> 
	<td width="6%" bgcolor="#0000FF">&nbsp;</td>
	<td bgcolor="#FF0000" ><div align="center"><strong><font face="verdana" size="3"></font></strong></div></td>
	<td bgcolor="#FF0000">&nbsp;</td>
</tr>
</table>
</html>


webs/index.php
<html>
  <head>
    <title>Tugas Besar Aplikasi Web Gis Peta Lokasi Sederhana</title>
  </head>
  <body>
    <td valign="top" bgcolor="#9ABC67" ><div align="center">
      <p><font face="comic sans" size="8" color="#000000">Menampilkan Peta</font> </p>
    <p>&nbsp;</p>
    </div
    </td>
    <br>
<td valign="top"><div align="justify"><font face="comic sans" size="5" color="#000000"><li><a href="http://localhost/cgi-bin/mapserv.exe?map=c:/ms4w/apps/bantul/bantul.map&layer=labelsmp&layer=smp_bantul&layer=bupati_poly_region&layer=JalanRuas_polyline&layer=Jalanutama_polyline&mode=map">Peta Bantul</a></li></font></div></td>
</br>
<br>
<td valign="top"><div align="justify"><font face="comic sans" size="5" color="#000000"><li><a href="http://localhost/cgi-bin/mapserv.exe?map=c:/ms4w/apps/bogor/bogor.map&mode=map">Peta Bogor</a></li></font> </div></td>
</br>
<br>
<td valign="top"><div align="justify"><font face="comic sans" size="5" color="#000000"><li><a href="http://localhost/cgi-bin/mapserv.exe?map=c:/ms4w/apps/map/peta.map&layer=District&mode=map">Peta Distrik</a></li></font> </div></td>
</br>
<p><a href="welcome.php">
  <input type="button" value="Back" name="back" onClick=" return cek()">
  </a></p>
  </body>
</body>
</html>


webs/signup.php
<?
if (isset($_POST['daftar']))
{
	include "./include/conn.php";
	$nama=ucwords($_POST['nama']);
	$user=$_POST['user'];
	$email=$_POST['email'];
	$password=md5($_POST['password']);
	
	if (!empty($nama) && !empty($user) && !empty($email) && !empty($password))
	{
		$cek=mysql_db_query($db,"select * from daftar where user='$user'",$koneksi);
		$valid=mysql_num_rows($cek);
		
		if ($valid){
			echo "<script> document.location.href='signup.php?status=Maaf, USERNAME Anda sudah ada yang punya!!'; </script>";
	} else {
		$insert=mysql_db_query($db,"insert into daftar values('','$nama','$user','$email','$password','$tanggal')",$koneksi);
		if ($insert)
		{
			echo "<script> document.location.href='signup.php?status=Selamat bergabung dengan situs kami.'; </script>";
	}
	}
	}
	else {
	echo "<script> document.location.href='signup.php?status=Maaf, Data yang anda kirim belum lengkap!!'; </script>";	
	}
}
?>


webs/welcome.php
<html>
<head><title>Welcome to Aplikasi Web GIS Menampilkan Peta Lokasi Sederhana</title></head>
<body>
<table>
<tr>
	<td>
	<blockquote>
	<font face="Arial, Helvetica, sans-serif" color="#000000" size="4"><h4>
	  <script language="JavaScript" type="text/javascript">
	  <!--
		var message="Selamat Datang Pada Aplikasi Web GIS Menampilkan Peta Lokasi Sederhana..."
		var neonbasecolor="white"
		var neontextcolor="red"
		var flashspeed=150  //dalam milisekon
		
		var n=0
		if (document.all||document.getElementById){
		document.write('<font color="'+neonbasecolor+'">')
		for (m=0;m<message.length;m++)
		document.write('<span id="neonlight'+m+'">'+message.charAt(m)+'</span>')
		document.write('</font>')
		}
		else
		document.write(message)
		
		function crossref(number){
		var crossobj=document.all? eval("document.all.neonlight"+number) : document.getElementById("neonlight"+number)
		return crossobj
		}
		
		
		if (n==0){
		for (m=0;m<message.length;m++)
		//eval("document.all.neonlight"+m).style.color=neonbasecolor
		crossref(m).style.color=neonbasecolor
		}
		
		
		crossref(n).style.color=neontextcolor
		
		if (n<message.length-1)
		n++
		else{
		n=0
		clearInterval(flashing)
		setTimeout("beginneon()",1500)
		return
		}
		}
		
		function beginneon(){
		if (document.all||document.getElementById)
		flashing=setInterval("neon()",flashspeed)
		}
		beginneon()
		//-->
		
</script>
	</h4></font>
	  <table>
	  <tr>
	  	<td valign="top"><div align="center"><font face="comic sans" size="4" color="#FF0000">Hello guys ... Anda berada di halaman utama web GIS untuk menampilkan peta lokasi. </font> </div></td>
		<td>&nbsp;</td>
	  </tr>
	  </table>
	  <p align="center">&nbsp;</p>
	  <p align="center"><br>
	    <img src="./img/GIS.jpg" border="0" title="Gambar Gis"></a></p>
	</blockquote>
	</td>
</tr>
<tr>
    </td>
	<td>
   	 <p><a href="about.php">
      <input type="button" value="About Us" name="About Us" onClick=" return cek()">
      </a></p>
	<p><a href="index.php">
      <input type="button" value="Next" name="next" onClick=" return cek()">
      </a></p>
    </td>
</tr>
</table>
</body>
</html>


webs/About.php
<? ?>
<html>
<head>
<title>Aplikasi Web GIS Menampilkan Peta Lokasi Sederhana</title>
<link rel="shortcut icon" href="./img/favicon.ico" type="image/x-icon">
</head>
<body>
  <center>
<table width="100%" cellspacing="8" cellpadding="2" >
<tr>
<td bgcolor="#FF0000" colspan="8"><div align="center">
  <font color="#FFFFFF" face="Times New Roman"><strong>About Us</strong></font></div>
</td>
<tr>
	<td valign="top"><p>&nbsp;</p>
	<p align="justify"><font size="2" face="Times New Roman">Hii Teman Semua...</font></p>
	<p align="justify"><font size="2" face="Times New Roman">Kita dari kelompok tugas besar GIS yang beranggotakan 4 orang mahasiswi yaitu:</font></p>
	<p align="justify"><font size="2" face="Times New Roman">1. Rosnani Nasrul dengan npm 1134015</font></p>
	<p align="justify"><font size="2" face="Times New Roman">2. Nur Mulyana Mustari dengan npm 1134047</font></p>
	<p align="justify"><font size="2" face="Times New Roman">3. Nur Fatimah Hasan dengan npm 1134016</font></p>
	<p align="justify"><font size="2" face="Times New Roman">4. Hariyati Suarni Nurindah dengan npm 1134021</font></p>
	<p align="justify"><font size="2" face="Times New Roman">Dari jurusan D4 Teknik Informatika kelas 3B dan kami berasaln dari daerah provinsi yang sama yaitu sulawesi selatan</font></p>
	<p align="justify"><font size="2" face="Times New Roman">Kami Berharap Semoga aplikasi web ini memberikan informasi yang bermanfaat bagi para pengguna dalam mengetahui peta lokasi pada web ini.</font></p>
	<p align="Center"><font size="2" face="Times New Roman">Terima Kasih Kunjungannya</font></p>
	<p><a href="welcome.php">
	<input type="button" value="Back" name="back" onClick=" return cek()">
      	</a></p>
	</td>
	</tr>
	</table>
	</center>
</body>
</html>
